test-integration/helpers/reading-builder.coffee

```diff
@@ -1,108 +1,122 @@
-    test.driver.findElement(css: '#reading-select').click()
-    test.waitAnd(css: '.select-reading-dialog:not(.hide)')
-    # Make sure nav bar does not cover buttons
-    test.scrollTop()
-
-    # Expand the chapter and then select the section
-    for section in sections
-      do (section) =>
-        section = "#{section}" # Ensure the section is a string so we can split it
-
-        # Selecting an entire chapter requires clicking the input box
-        # So handle chapters differently
-        isChapter = not /\./.test(section)
-        if isChapter
-          test.waitClick(css: ".dialog:not(.hide) [data-chapter-section='#{section}'] .chapter-checkbox input")
-        else
-          # BUG? Hidden dialogs remain in the DOM. When searching make sure it is in a dialog that is not hidden
-          test.driver.findElement(css: ".dialog:not(.hide) [data-chapter-section='#{section}']").isDisplayed().then (isDisplayed) =>
-            # Expand the chapter accordion if necessary
-            unless isDisplayed
-              test.waitClick(css: ".dialog:not(.hide) [data-chapter-section='#{section.split('.')[0]}']")
-
-            test.waitClick(css: ".dialog:not(.hide) [data-chapter-section='#{section}']")
-
-    if verifyAddReadingsDisabled
-      # Verify "Add Readings" is disabled and click Cancel
-      test.waitAnd(css: '.dialog:not(.hide) .-show-problems[disabled]')
-      test.waitClick(css: '.dialog:not(.hide) .panel-footer [aria-role="close"]')
-      # Confirm the "Unsaved Changes" dialog
-      test.waitClick(css: '.-tutor-dialog-parent .tutor-dialog.modal.fade.in .modal-footer .ok.btn')
-      test.sleep(1000) # Wait for dialog to close
-    else
-      # Click "Add Readings"
-      test.waitClick(css: '.-show-problems') # BUG: wrong class name
-
-  switch action
-    when 'PUBLISH'
-      # Wait up to 3min for publish to complete
-      test.waitClick(css: '.async-button.-publish')
-      Calendar.verify(test, 3 * 60 * 1000)
-
-    when 'SAVE' then test.waitClick(css: '.async-button.-save')
-    when 'CANCEL'
-      # BUG: "X" close button behaves differently than the footer close button
-      test.waitClick(css: '.footer-buttons [aria-role="close"]')
-      # BUG: Should not prompt when canceling
-      # Confirm the "Unsaved Changes" dialog
-      test.waitClick(css: '.-tutor-dialog-parent .tutor-dialog.modal.fade.in .modal-footer .ok.btn')
-      test.sleep(1000) # Wait for dialog to close
-      Calendar.verify(test)
-
-    when 'DELETE'
-      # Wait up to 60sec for delete to complete
-      test.waitClick(css: '.async-button.delete-link')
-      # Accept the browser confirm dialog
-      test.driver.wait(selenium.until.alertIsPresent()).then (alert) ->
-        alert.accept()
-
-      Calendar.verify(test, 60 * 1000)
-
-
-module.exports = {setDate, edit}
+  # Helper methods for dealing with the Reading Assignment Builder
+class ReadingBuilder extends TestHelper
+
+  constructor: (test) ->
```

Comment:
Why do these helpers take `test` instead of driver? If it is only because of the helpers in `describe.coffee` then would it make sense to pull them out of describe and revisit all these places?